### PR TITLE
login: do not issue wall messages on ptys for suspend and hibernate

### PR DESCRIPTION
**Pull request by cgzones**
Fixes: #23520

### CODE REVIEW

File: src/login/logind-utmp.c [hunk 1]
```diff
@@ -47,6 +47,15 @@ bool logind_wall_tty_filter(const char *tty, void *userdata) {
...
 
         assert(m);
+        assert(m->scheduled_shutdown_action);
+
+        if (IN_SET(m->scheduled_shutdown_action->handle,
+                HANDLE_SUSPEND,
+                HANDLE_HIBERNATE,
+                HANDLE_HYBRID_SLEEP,
+                HANDLE_SUSPEND_THEN_HIBERNATE) &&
+                path_startswith(tty, "/dev/pts/"))
```
Comment (keszybz) [inline src/login/logind-utmp.c:57]: Indentation is borked, which makes this hard to read: please align `HANDLE_*` to the same column as `m->`.

**Review by keszybz — COMMENTED** [review]
Hmmm. Hmmm. I had to reread all the code that sets and uses `scheduled_shutdown_tty` to figure out what is going on here. IIUC, the idea is that we record the tty where the command to shut down / hibernate / etc. originated. When doing the wall message, we loop over all utmp entries, and send the message to terminals where users are logged in, except for the originating tty.

This patch "fixes" the issue by dropping the notifications on pseudo-ttys, but I don't think this makes much sense. In particular, it'll drop notifications for ssh sessions, which would seem to be quite useful and which we should keep.

It seems that the patch which enabled wall messages by default exposed a latent bug. People weren't generally requesting notifications, so nobody noticed that they were being delivered improperly. But if somebody requested them, the same issue would appear.

The question instead is: why do people have their pseudoterminal sessions registered in utmp? On my machine, `utmpdump </var/run/utmp` yields `tty2` for my graphical session, and `pts/12` for an ssh login. So I see the wall message exactly once, on the ssh pty.

**Review by pieniacy — COMMENTED** [review]
IMO any hardcoded mechanism is not a solution to this problem, this should be configurable at least with respect to suspend/hibernate and shutdown/reboot.

See https://github.com/systemd/systemd/issues/23520#issuecomment-1140206521

**Comment by keszybz** [comment]
After looking at this even more, I don't think we should take this approach. As described above, this patch hides the issue, but it makes the logic even more accidental. I think it's fair to say that the logic of using utmp lists for wall messages has severe problems. I think we should revert https://github.com/systemd/systemd/commit/cdf370626f08ed509a5dde9d5618eed29d625032 instead. I'll submit a PR with a proper justification there.

**Pull request closed without merging**